llvm/lib/CheerpUtils/GlobalDepsAnalyzer.cpp

```diff
@@ -124,12 +125,35 @@ void GlobalDepsAnalyzer::extendLifetime(Function* F)
 	assert( visited.empty() );
 }
 
+static void replaceFunctionAliasWithAliasee(llvm::Module &module, StringRef name) {
+	GlobalAlias *funcAlias = module.getNamedAlias(name);
+	if (funcAlias) {
+		Function *actualFunc = dyn_cast<Function>(funcAlias->getAliaseeObject());
+
+		assert(actualFunc);
+
+		funcAlias->replaceAllUsesWith(actualFunc);
```

Comment:
Can't we just replace with the newFunc as we do below? why this intermediate replacement?

Reply:
This seems to also work, currently running the tests

Reply:
I thought it was necessary so I could make it have the same name

Reply:
Yeah, that totally works